api/workload/v2/reader.go

```diff
@@ -0,0 +1,94 @@
+package workload
+
+import (
+	"context"
+	"sync"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/spire/proto/api/workload"
+)
+
+type streamReader struct {
+	Chan          chan *workload.X509SVIDResponse
+	logger        *logrus.Logger
+	streamManager *streamManager
+
+	// atomic stream
+	mu     *sync.RWMutex
```

Comment:
I assume this struct is never copied (you are currently returning a pointer).

No need to make this a pointer, the zero value is usable so if it's not a pointer you don't need to initialize it.

Reply:
i think i was being overly cautious, agree its fine to use the zero value